torch/lib/THD/base/ChannelUtils.hpp

```diff
@@ -192,4 +187,22 @@ void send_value(int socket, T&& value, bool more_data = false) {
   send_bytes<T>(socket, &value, 1, more_data);
 }
 
+class ResourceGuard {
```

Comment:
It is okay for me to use `class` instead of `struct` but just wanted to know if this is on purpose or not.